include/deal.II/base/tensor.h

```diff
@@ -938,23 +928,16 @@ Tensor<rank_,dim,Number>::operator[] (const TableIndices<rank_> &indices)
 
 
 template <int rank_, int dim, typename Number>
-inline
-Tensor<rank_,dim,Number> &
-Tensor<rank_,dim,Number>::operator = (const Tensor<rank_,dim,Number> &t)
-{
-  for (unsigned int i=0; i<dim; ++i)
-    values[i] = t.values[i];
-  return *this;
-}
-
-
-template <int rank_, int dim, typename Number>
 template <typename OtherNumber>
 inline
 Tensor<rank_,dim,Number> &
 Tensor<rank_,dim,Number>::operator = (const Tensor<rank_,dim,OtherNumber> &t)
 {
-  for (unsigned int i=0; i<dim; ++i)
+  // avoid a warning with icc that complains about an empty loop. Even in
+  // case of dim==0 we have an element available (as a workaround for
+  // another set of warnings about an empty c-style array *sigh).
+  // If dim==0, ust copy this bogus element:
```

Comment:
I will admit that I'm not fond of this solution because it uses implicit knowledge from one place in another place. That seems wrong and not long-term maintainable.

What is wrong with providing the explicit specializations of this function for rank=1,2,3?